src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,173 @@
+            Node<T> nodeIndex = nodeBy(index);
+            nodeIndex.prev = new Node<>(nodeIndex.prev, value, nodeIndex);
+            currentAmount++;
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        int previousAmount = currentAmount;
+        for (T value : list) {
+            addElement(value);
+        }
+        return currentAmount - previousAmount == list.size();
     }
 
     @Override
     public T get(int index) {
-        return null;
+        check(index);
+        return nodeBy(index).getData();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        check(index);
+        Node<T> nodeIndex = nodeBy(index);
+        T oldData = nodeIndex.getData();
+        nodeIndex.setData(value);
+        return oldData;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        check(index);
+        Node<T> removableNode = nodeBy(index);
+        T data = removableNode.data;
+        if (index == 0) {
+            if (currentAmount == 1) {
+                first = null;
+                last = null;
+            } else {
+                first = removableNode.next;
+                removableNode.next.prev = null;
+            }
+        } else if (index > 0 && index < currentAmount - 1) {
+            removableNode.prev.next = removableNode.next;
+            removableNode.next.prev = removableNode.prev;
+        } else if (index == currentAmount - 1) {
+            last = removableNode.prev;
+            last.next = null;
+        }
+        currentAmount--;
+        return data;
     }
 
     @Override
     public boolean remove(T t) {
+        for (int index = 0; index < currentAmount; index++) {
+            Node<T> removableNode = nodeBy(index);
+            if (t == removableNode.data
+                    || t != null && t.equals(removableNode.data)) {
+                if (index != 0) {
+                    removableNode.prev.next = removableNode.next;
+                }
+                if (index != currentAmount - 1) {
+                    removableNode.next.prev = removableNode.prev;
+                }
+                currentAmount--;
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return currentAmount;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return currentAmount == 0;
+    }
+
+    private static class Node<T> {
+
+        T data;
+        Node<T> next;
+        Node<T> prev;
+
+        private Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.data = element;
+            this.next = next;
+        }
+
+        private void setData(T value) {
+            data = value;
+        }
+
+        private T getData() {
+            return data;
+        }
+    }
+
+    private Node<T> nodeBy(int index) {
```

Comment:
is that correct method naming? 
https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names